wger.py

```diff
@@ -42,62 +52,86 @@ def post_info(self,url_file,post_data):
-        return "Item Deleted"
+        response=requests.delete(url=delete_url,headers={'Authorization': self.auth_token})
+        return response
+
+    def match(self,dict1,dict_item,param1,param2):
+        current_value=""
+        for i in dict1:
+            if i[param1] == dict_item:
+                current_value = i.get(param2)
+        return current_value
 
-    def chose_weight(self,date,weight):
+
+    def weight_entry(self,date,weight):
         """The function creates weight goals for a specific date."""
 
-        data={ "id":"","date":date,"weight":weight,"user":""}
-        self.post_info('weightentry/',data)
-        return "Weight goal added successfully"
+        data={ "id":"",
+               "date":date,
+               "weight":weight,
+               "user":""}
+        response=self.post_info('weightentry/',data)
+        return response
     
     def make_plan(self,plan_name):
         """The function creates a new nutrition plan and also a meal for the plan"""
 
-        plan_data = {"id": "", "creation_date": "", "description": plan_name, "has_goal_calories": True,"language": 2}
+        plan_data = {"id": "",
+                     "creation_date": "",
+                     "description": plan_name,
+                     "has_goal_calories": True,
+                     "language": 2}
         plan=self.post_info('nutritionplan/',plan_data)
-        meal_data={"id":"","plan": plan['id'],"order": "","time":""}
+        meal_data={"id":"",
+                   "plan": plan.get('id'),
+                   "order": "",
+                   "time":""}
         meal=self.post_info('meal/',meal_data)
         return meal
 
-    def add_item(self,meal):
+    def add_item(self,meal,amount):
         """The function adds a new item to a specific meal"""
 
         items=self.get_info('/ingredient')
         new_item=random.choice(items['results'])
-        ingredient=new_item['id']
-        data_item={"id":"","meal": meal,"ingredient": ingredient,"weight_unit":"","order": 1,"amount":"100"}
+        ingredient=new_item.get('id')
+        data_item={"id":"",
+                   "meal": meal,
+                   "ingredient": ingredient,
+                   "weight_unit":"",
+                   "order": 1,
+                   "amount":amount}
         self.post_info('mealitem/',data_item)
-        return new_item['energy']
+        return new_item.get('energy')
 
     def week_program(self,kcals_1,kcals_2):
         """The function creates nutrition plans for every single day in a week based on energy targets"""
 
         days=['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
         for i in days :
             nutrition_plan=self.make_plan(i)
-            meal_id=nutrition_plan['id']
+            meal_id=nutrition_plan.get('id')
             total_kcals=0
             total_items=0
             if i in ['Monday', 'Tuesday', 'Wednesday', 'Thursday']:
                 while(total_kcals<=kcals_1):
```

Comment:
The conditions on lines 84 and 89 are basically the same. You need a different logic in order to add to a specific plan until you reach the closest point to kcals_1(without going over it) and to add to a specific plan until you go over kcals_2.